nimble/controller/src/ble_ll_adv.c

```diff
@@ -3098,6 +3121,42 @@ ble_ll_adv_reschedule_event(struct ble_ll_adv_sm *advsm)
 #endif
 }
 
+static void
+ble_ll_adv_update_adv_scan_rsp_data(struct ble_ll_adv_sm *advsm)
+{
+    if (!(advsm->flags & BLE_LL_ADV_SM_FLAG_NEW_ADV_DATA) &&
+                    !(advsm->flags & BLE_LL_ADV_SM_FLAG_NEW_SCAN_RSP_DATA)) {
+        return;
+    }
+
+#if MYNEWT_VAL(BLE_LL_CFG_FEAT_LL_EXT_ADV)
+    if (advsm->aux_active) {
+        return;
+    }
+#endif
+
+    if (advsm->flags & BLE_LL_ADV_SM_FLAG_NEW_ADV_DATA){
+
+        if (advsm->updated_data) {
```

Comment:
why this `if` is needed here? flag should be enough here and also it's not present below

Reply:
it is not enough for operation `Unchanged data (just update the Advertising DID)(0x04)`